### PR TITLE
ARROW-16406: [Docs][R] Update documentation with new nightly location

### PR DESCRIPTION
**Pull request by assignUser**
merge after #13149 and #13241

### CODE REVIEW

File: r/vignettes/developers/setup.Rmd [hunk 1]
```diff
@@ -47,13 +47,8 @@ recent version of the library without building from source.
 On Linux, you can download a .zip file containing libarrow from the
 nightly repository.
 
-To see what nightlies are available, you can use arrow's (or any other S3 client's) S3 listing functionality to see what is in the bucket `s3://arrow-r-nightly/libarrow/bin`:
+To see what nightlies are available, you can visit the [nightly build server](https://nightlies.apache.org/arrow/r/libarrow/bin). Version numbers in that repository correspond to dates. See the [distro map](https://raw.githubusercontent.com/apache/arrow/master/r/tools/distro-map.csv) for compatibility between binaries and OSs.
```
Comment (nealrichardson) [inline r/vignettes/developers/setup.Rmd:50]: distro-map.csv is still not present in this repo

**Review by nealrichardson — CHANGES_REQUESTED** [review]
Also, all of these links 404, is this expected?

File: r/vignettes/developers/setup.Rmd (hunk 1 above)
Comment (assignUser) [reply r/vignettes/developers/setup.Rmd:50]: #13241 adds it

File: r/vignettes/developers/install_details.Rmd [hunk 2]
```diff
@@ -100,7 +100,7 @@ The libarrow binaries will only be retrieved if you have set the environment var
 
 If found, they will be downloaded and bundled when your R package compiles.
 For a list of supported distributions and versions,
-see the [arrow-r-nightly](https://github.com/ursa-labs/arrow-r-nightly/blob/master/README.md) project.
+see the [nightly build server](https://nightlies.apache.org/arrow/r/README).
```
Comment (nealrichardson) [inline r/vignettes/developers/install_details.Rmd:103]: This URL is 404. Might want to point this somewhere else, or delete entirely--we have an open issue to change how this works anyway.

File: r/vignettes/developers/setup.Rmd (hunk 1 above)
Comment (nealrichardson) [reply r/vignettes/developers/setup.Rmd:50]: Is there a good reason that PR is separate from this? Seems odd to merge either of them separately.

**Review by nealrichardson — COMMENTED** [review]
Looks like you have some merge conflicts, and I recommend merging https://github.com/apache/arrow/pull/13241 into this one and resolving them together.

**Comment by assignUser** [comment]
Task merged with #13241/ARROW-16405

**Pull request closed without merging**